src/pymorize/generic.py

```diff
@@ -252,3 +253,35 @@ def trigger_compute(data, rule_spec, *args, **kwargs):
         return data.compute()
     # Data doesn't have a compute method, do nothing
     return data
+
+
+def sort_dimensions(data, rule_spec):
+    """
+    Sorts the dimensions of a DataArray based on the array_order attribute of the
+    rule_spec. If the array_order attribute is not present, it is inferred from the
+    dimensions attribute of the data request variable.
+    """
+    missing_dims = rule_spec.get("sort_dimensions_missing_dims", "raise")
+
+    if hasattr(rule_spec, "array_order"):
+        array_order = rule_spec.array_order
+    else:
+        dimensions = rule_spec.data_request_variable.dimensions
+        # Pattern to match a valid array_order (e.g. "time lat lon", but not
+        # "[time lat lon]" or "time,lat,lon")
+        pattern = r"^(?!\[.*\]$)(?!.*,.*)(?:\S+\s*)+$"
+        if isinstance(dimensions, str) and re.fullmatch(pattern, dimensions):
+            array_order = dimensions.split(" ")
+        elif isinstance(dimensions, list) or isinstance(dimensions, tuple):
+            array_order = dimensions
+        else:
+            logger.error(
+                "Invalid dimensions in data request variable: "
+                f"{rule_spec.data_request_variable}"
+            )
+            raise ValueError("Invalid dimensions in data request variable")
```

Comment:
I would raise a `TypeError` here, not a `ValueError`, since the checks above are determining if the array dimensions are list-ish (either as a string with split, or as a tuple/list).

```suggestion
            raise TypeError("Invalid dimensions in data request variable")
```